solrmonitor/zkdispatcher.go

```diff
@@ -83,20 +82,13 @@ func NewZkDispatcher(logger zk.Logger) *ZkDispatcher {
 }
 
 func (d *ZkDispatcher) Close() {
-	if !atomic.CompareAndSwapInt32(&d.closed, 0, 1) {
```

Comment:
Yeah, I had thought about this, too. At first, I formulated with the CAS because I thought we allowed repeated `Close`. I then realized that the code previously panic'ed. So this final formulation seems a bit redundant since the line below will panic on its own...